components/tableIndexer/tables_cropper.py

```diff
@@ -0,0 +1,180 @@
+    )
+    parser.add_argument(
+        "--json-path",
+        dest="json_path",
+        required=True,
+        help="Specify the path of the json file containing the scan results",
+    )
+
+    parser.add_argument(
+        "--log-level",
+        dest="log_level",
+        default="INFO",
+        help="Set the logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)",
+    )
+
+    return parser.parse_args()
+
+
+class TableCropper:
+    def __init__(
+        self,
+        img_path,
+        json_path,
+        tokens=[],
+        class_thresholds={"table": 0.5, "table rotated": 0.5, "no object": 10},
+        padding=10,
+        output_dir="./outputs/tables",
+    ):
+        self.img_path = img_path
+        self.json_path = json_path
+        self.tokens = tokens
+        self.class_thresholds = class_thresholds
+        self.padding = padding
+        self.output_dir = output_dir
+
+    def crop_table(self, objects):
+        try:
+            logging.info("✨ Sentinel Table Cropper has started...")
+            table_crops = []
+            for obj in objects:
+                if obj["score"] < self.class_thresholds[obj["label"]]:
+                    continue
+
+                cropped_table = {}
+
+                bbox = obj["bbox"]
+                bbox = [
+                    bbox[0] - self.padding,
+                    bbox[1] - self.padding,
+                    bbox[2] + self.padding,
+                    bbox[3] + self.padding,
+                ]
+                image = Image.open(self.img_path).convert("RGB")
+                cropped_img = image.crop(bbox)
+
+                table_tokens = [
+                    token
+                    for token in self.tokens
+                    if self.iob(token["bbox"], bbox) >= 0.5
+                ]
+                for token in table_tokens:
+                    token["bbox"] = [
+                        token["bbox"][0] - bbox[0],
+                        token["bbox"][1] - bbox[1],
+                        token["bbox"][2] - bbox[0],
+                        token["bbox"][3] - bbox[1],
+                    ]
+
+                # If table is predicted to be rotated, rotate cropped image and tokens/words:
+                if obj["label"] == "table rotated":
+                    cropped_img = cropped_img.rotate(270, expand=True)
+                    for token in table_tokens:
+                        bbox = token["bbox"]
+                        bbox = [
+                            cropped_img.size[0] - bbox[3] - 1,
+                            bbox[0],
+                            cropped_img.size[0] - bbox[1] - 1,
+                            bbox[2],
+                        ]
+                        token["bbox"] = bbox
+
+                cropped_table["image"] = cropped_img
+                cropped_table["tokens"] = table_tokens
+
+                table_crops.append(cropped_table)
+
+            return table_crops
+        except KeyboardInterrupt:
```

Comment:
**Question: What is the benefit of catching SIGINTERUPT here?**

If you arne't cleaning up any resources here its not worth catching and rethrowing the exception imho